### PR TITLE
Release 1.8.0

### LINKED ISSUE #75: Write PowerShell profile customizations to $Profile.CurrentUserAllHosts
There may be other issues, but the most glaring difference is that the cookbook only puts the profile in as the PowerShell profile, which is 1 of 6 profiles (and not the one used by ISE).

We need to review how profiles are selected, how they work with ISE, and make changes to support PowerShell ISE as a first-class option for installation.

### LINKED ISSUE #130: Move PowerShell Profile customizations to a managed file
Right now, we use :create_if_missing on the PowerShell profile template, which makes it hard to update. A better way would be to move all our customizations to an external file (perhaps a Module) and then include that external file into profile with something like the line cookbook.

### PR DESCRIPTION
**Pull request by dougireton**
Fix #130
  Move PowerShell Profile customizations to a PowerShell module on the $env:PSModulePath

Fix #75
  Write PowerShell profile customizations to $Profile.CurrentUserAllHosts

### CODE REVIEW

File: recipes/windows.rb [hunk 1]
```diff
@@ -27,6 +27,7 @@
 packages = node['chefdk_bootstrap']['package']
 
 packages.each do |pkg, install|
+  Chef::Log.info "#{pkg} is #{install}"
```
Comment (oifland) [inline recipes/windows.rb]: This can be removed

File: .kitchen.yml [hunk 2]
```diff
@@ -11,11 +11,19 @@ platforms:
...
     run_list:
       - recipe[win_proxy]
       - recipe[chefdk_bootstrap]
-
-verifier:
-  name: inspec
+    attributes:
+      chefdk_bootstrap:
+        package:
+          atom: false,
```
Comment (oifland) [inline .kitchen.yml]: I think the comma is turning this into a string

File: templates/default/chefdk_bootstrap.psm1.erb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<% if node['chefdk_bootstrap']['proxy']['http'] -%>
```
Comment (oifland) [inline templates/default/chefdk_bootstrap.psm1.erb]: Apache 2.0 header

File: test/integration/default/git_spec.rb [hunk 4]
```diff
@@ -12,28 +12,28 @@
...
-describe command("& \"#{git_credential_mgr}\" version") do
-  its(:exit_status) { should eq(0) }
-  its(:stderr) { should eq '' }
-end
-
-# PoshGit
-describe powershell('test-path Function:\PoshGitPrompt') do
-  its(:stdout) { should match(/^True\R/) }
-end
+# describe command('git --version') do
```
Comment (oifland) [inline test/integration/default/git_spec.rb]: Commented out?

File: test/integration/default/conemu_spec.rb [hunk 5]
```diff
@@ -12,6 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-describe command('C:\Program Files\ConEmu\ConEmu\ConEmuC.exe') do
-  its(:stderr) { should match(/ConEmu/) }
-end
+# describe command('C:\Program Files\ConEmu\ConEmu\ConEmuC.exe') do
```
Comment (oifland) [inline test/integration/default/conemu_spec.rb]: More commented code

File: recipes/powershell_profile.rb [hunk 6]
```diff
@@ -13,12 +13,30 @@
...
+
+current_user_all_hosts_profile = win_friendly_path(File.join(Dir.home, 'Documents\WindowsPowerShell\Profile.ps1'))
 
-template powershell_profile do
+file current_user_all_hosts_profile do
   action :create_if_missing
-  source 'global_profile.ps1.erb'
+end
+
+append_if_no_line 'Setup ChefDK environment for PowerShell' do
```
Comment (oifland) [inline recipes/powershell_profile.rb]: Roll these up into a single call into our module

**Comment by dougireton** [comment]
@oifland kitchen verify passes. This is ready to be merged.

File: recipes/powershell_profile.rb [hunk 7]
```diff
@@ -13,12 +13,28 @@
...
-template powershell_profile do
+directory current_user_all_hosts_profile_dir
+
+file current_user_all_hosts_profile do
   action :create_if_missing
-  source 'global_profile.ps1.erb'
+end
+
+append_if_no_line 'Set proxy env vars in Current User profile' do
+  path lazy { current_user_all_hosts_profile }
```
Comment (oifland) [inline recipes/powershell_profile.rb]: I'm puzzled on the need for lazy here. That's just a local variable from above...

**Comment by oifland** [comment]
Once this is released, I want to test out the upgrade experience before we tweet. = )

**Comment by dougireton** [comment]
Fair enough

**Pull request merged**